sys/net/gnrc/netif/init_devs/auto_init_at86rf2xx.c

```diff
@@ -52,22 +56,24 @@ void auto_init_at86rf2xx(void)
     for (unsigned i = 0; i < AT86RF2XX_NUM; i++) {
         LOG_DEBUG("[auto_init_netif] initializing at86rf2xx #%u\n", i);
 
-        at86rf2xx_setup(&at86rf2xx_devs[i], &at86rf2xx_params[i], i);
+        at86rf2xx_init_event(&at86rf2xx_bhp[i], &at86rf2xx_params[i], &at86rf2xx_netdev[i].submac.dev, EVENT_PRIO_HIGHEST);
```

Comment:
We should check for the return value here, to avoid a crash when the driver init fails, in case of a hardware fault or misconfiguration. For example:
```C
        int init;
        if ((init = at86rf2xx_init_event(&at86rf2xx_bhp[i], &at86rf2xx_params[i],
                                         &at86rf2xx_netdev[i].submac.dev, EVENT_PRIO_HIGHEST))) {
            LOG_ERROR("at86rf2xx #%u init failed: %i\n", i, init);
            continue;
        }
```

Similar for `lwip`, `openthread`, `openwsn`, I think.